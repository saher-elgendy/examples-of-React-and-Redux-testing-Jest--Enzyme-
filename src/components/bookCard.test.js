import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import BookCard from './bookCard';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import './../setupTests';


const mockStore = configureStore([]);

describe('BookStore component', () => {
    let wrapper;
    let mockProps;
    let store;
   
    beforeEach(() => {
        store = mockStore({
            bag: [],
        })

        mockProps = {
            bag: [],
            book: {},
            addBookToBag: jest.fn(),
            removeBookFromBag: jest.fn()
        }

        wrapper = mount(
            <Provider store={store}>
                <BookCard {...mockProps} />
            </Provider>
        );

    })

    it('render BookCard component correctly', () => {
        expect(mountToJson(wrapper)).toMatchSnapshot();
    })

    it('shows book details correctly', () => {
        mockProps = {
            ...mockProps,
            book: {
                name: 'The secret',
                author: 'Ronda Bayrne',
                pagesNumber: 300,
            }
        }

        wrapper = mount(
            <Provider store={store}>
                <BookCard {...mockProps} />
            </Provider>
        );
        const bookName = wrapper.find('#book-name')
        const author = wrapper.find('#book-author')
        const pageCount = wrapper.find('#page-count')
        expect(bookName.text()).toBe('The secret')
        expect(author.text()).toBe('Author: Ronda Bayrne')
        expect(pageCount.text()).toBe('Page Count: 300')
    });
});