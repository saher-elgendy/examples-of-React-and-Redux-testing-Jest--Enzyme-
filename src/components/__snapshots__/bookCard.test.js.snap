// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookStore component render BookCard component correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(BookCard)
    addBookToBag={[MockFunction]}
    bag={Array []}
    book={Object {}}
    removeBookFromBag={[MockFunction]}
  >
    <BookCard
      addBookToBag={[Function]}
      bag={Array []}
      book={Object {}}
      removeBookFromBag={[Function]}
    >
      <li
        className="book-card"
      >
        <img
          alt="book"
          className="book-cover"
        />
        <h4
          id="book-name"
        />
        <h4
          id="book-author"
        >
          Author: 
          <span
            style={
              Object {
                "fontWeight": "normal",
              }
            }
          />
        </h4>
        <h4
          id="page-count"
        >
          Page Count: 
          <span
            style={
              Object {
                "fontWeight": "normal",
              }
            }
          />
        </h4>
        <BookCardButton
          addBookToBag={[Function]}
          bag={Array []}
          book={Object {}}
          removeBookFromBag={[Function]}
        >
          <button
            className="book-card-btn add-book-btn"
            onClick={[Function]}
          >
            ADD TO BAG
          </button>
        </BookCardButton>
      </li>
    </BookCard>
  </Connect(BookCard)>
</Provider>
`;
